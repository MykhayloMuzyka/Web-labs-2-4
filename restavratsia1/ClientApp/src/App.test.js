import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import Footer from "./components/footer";
import request from "./components/Utils/RequestWrapper";
import ProfilePage from "./components/ProfilePage";

it("server interaction by getting orders", async () => {
  const result = await request({
    method: "get",
      url: "ads/get_all/5c76f2b1-7860-4166-8179-4bf6da0ed116c/",
  });
  expect(result.data.value).toBeTruthy();
});

it("correct component rendering", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("userOffice renders as expected", () => {
    let elem = create(<ProfilePage />);
    expect(elem).toMatchSnapshot();
});