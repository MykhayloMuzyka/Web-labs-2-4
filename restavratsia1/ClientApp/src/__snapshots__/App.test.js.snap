// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userOffice renders as expected 1`] = `
<div>
  <div
    className="d-flex-spacebtw"
  >
    <h1
      className="text-default ml-3 mb-1 mt-1"
      id="profile-title"
    >
      Profile
    </h1>
    <div
      className="mr-3 flex-row-center"
    >
      <div
        className="mr-1 ml-1"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            delete
          </span>
        </span>
      </div>
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Edit
        </span>
      </span>
    </div>
  </div>
  <form
    className="d-flex flex-dir-column align-items-center"
    onSubmit={[Function]}
  >
    <div
      className="col-12 mb-2 flex-center"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-1"
      >
        <img
          alt="profile photo"
          className="MuiAvatar-img"
          src="ava.jpg"
        />
      </div>
    </div>
    <div
      className="col-7 p-4 flex-column-center"
      id="profile"
    >
      <div
        className="col-12 flex-column col-md-12"
      >
        <h3
          className="profile-chapters text-default mb-1 mt-1"
        >
          Account info
        </h3>
        <div
          className="MuiFormControl-root MuiTextField-root col-12"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
              disabled={true}
              id="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="mt-2"
        >
          <div
            className="d-none"
            id="change-pass-btn"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                change password
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-12 flex-column col-md-12"
      >
        <h3
          className="profile-chapters text-default mt-4 mb-1"
        >
          Personal info
        </h3>
        <div
          className="flex-column col-12"
        >
          <div
            className="MuiFormControl-root MuiTextField-root col-12"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
              data-shrink={false}
              htmlFor="companyName"
              id="companyName-label"
            >
              Company Name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                disabled={true}
                id="companyName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root col-12"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
              disabled={true}
              id="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root col-12"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
            data-shrink={false}
            htmlFor="phone"
            id="phone-label"
          >
            Phone
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
              disabled={true}
              id="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-container col-10 MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            className="MuiPaper-root MuiAlert-root MuiAlert-standardError col-12 MuiPaper-elevation0"
            role="alert"
          >
            <div
              className="MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              className="MuiAlert-message"
            >
              Password change failed! 
              
            </div>
            <div
              className="MuiAlert-action"
            >
              <button
                aria-label="close"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex-row-center mb-1 mt-3"
    >
      <div
        className="d-none"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            cancel
          </span>
        </button>
      </div>
      <div
        className="d-none"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            confirm
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
